app/models/service_template_provision_task.rb

```diff
@@ -158,7 +155,7 @@ def deliver_to_automate(req_type = request_type, _zone = nil)
   end
 
   def resource_action
-    source.resource_actions.detect { |ra| ra.action == 'Provision' } if source.respond_to?(:resource_actions)
+    @resource_action ||= source.resource_actions.find_by(:action => 'Provision') if source.respond_to?(:resource_actions)
```

Comment:
so the old code was written so all loaded resource_actions can be searched
The new code will create a query

probably doesn't matter

old code would use resource_actions cache. (but keep searching) and the new code does need the `@ ||=` b/c it would do the query over and over

Reply:
Followup on my previous thought. (this still doesn't mean we need to do anything about this comment)

If we think the resource_actions will be pre-loaded, we could always do the `source.resource_action.loaded?` trick, but that gets verbose.